dev-libs/boost/boost-1.88.0.ebuild

```diff
@@ -0,0 +1,480 @@
+	b2 "${b2_opts[@]}"
+}
+
+src_configure() {
+	# Workaround for too many parallel processes requested, bug #506064
+	[[ "$(makeopts_jobs)" -gt 64 ]] && MAKEOPTS="${MAKEOPTS} -j64"
+
+	# We don't want to end up with -L/usr/lib on our linker lines
+	# which then gives us lots of
+	#   skipping incompatible /usr/lib/libc.a when searching for -lc
+	# warnings
+	[[ -n ${ESYSROOT} ]] && local icuarg="-sICU_PATH=${ESYSROOT}/usr"
+
+	OPTIONS=(
+		$(usex debug gentoodebug gentoorelease)
+		"-j$(makeopts_jobs)"
+		-q
+		-d+2
+		pch=off
+		$(usex icu "${icuarg}" '--disable-icu boost.locale.icu=off')
+		$(usev !mpi --without-mpi)
+		$(usev !nls --without-locale)
+		$(usev !context '--without-context --without-coroutine --without-fiber')
+		$(usev !stacktrace --without-stacktrace)
+		--boost-build="${BROOT}"/usr/share/b2/src
+		--layout=system
+		# building with threading=single is currently not possible
+		# https://svn.boost.org/trac/boost/ticket/7105
+		threading=multi
+		link=shared
+		# this seems to be the only way to disable compression algorithms
+		# https://www.boost.org/doc/libs/1_70_0/libs/iostreams/doc/installation.html#boost-build
+		-sNO_BZIP2=$(usex bzip2 0 1)
+		-sNO_LZMA=$(usex lzma 0 1)
+		-sNO_ZLIB=$(usex zlib 0 1)
+		-sNO_ZSTD=$(usex zstd 0 1)
+		boost.stacktrace.from_exception=off
+	)
+
+	if [[ ${CHOST} == *-darwin* ]]; then
+		# We need to add the prefix, and in two cases this exceeds, so prepare
+		# for the largest possible space allocation.
+		append-ldflags -Wl,-headerpad_max_install_names
+	fi
+
+	# Use C++17 globally as of 1.80
+	append-cxxflags -std=c++17
+
+	if [[ ${CHOST} != *-darwin* ]]; then
+		# On modern macOS, file I/O is already 64-bit by default,
+		# there's no support for special options like O_LARGEFILE.
+		# Thus, LFS must be disabled.
+		#
+		# On other systems, we need to enable LFS explicitly for 64-bit
+		# offsets on 32-bit hosts (#894564)
+		append-lfs-flags
+	fi
+}
+
+multilib_src_compile() {
+	ejam \
+		--prefix="${EPREFIX}"/usr \
+		"${OPTIONS[@]}" || die
+
+	if multilib_native_use tools; then
+		pushd tools >/dev/null || die
+		ejam \
+			--prefix="${EPREFIX}"/usr \
+			"${OPTIONS[@]}" \
+			|| die "Building of Boost tools failed"
+		popd >/dev/null || die
+	fi
+}
+
+multilib_src_test() {
+
+	##
+	## Preparation
+	##
+
+	# Some test suites have no main because normally boost.test can
+	# automatically initialize & run them, but this only seems to be
+	# supported for statically linked builds/tests.
+	# Therefore we use an explicit list of tests which need patching
+	# with an additional main().
+	# Determining this dynamically is not really possible.
+	local libs_needpatch=(
+		"accumulators"
+	)
+
+	einfo "Patching:" ${libs_needpatch[@]}
+
+	local lib
+	for lib in "${libs_needpatch[@]}"; do
+		# move into library test dir
+		pushd "${BUILD_DIR}/libs/${lib}/test" >/dev/null || die
+			# find all test cases and patch them
+			local testcases testcase
+			readarray -t testcases < <(find . -name "*.cpp")
```

Comment:
This should be using `readarray -d''` and `-print0` in `find` to be safe w/ dodgy filenames.

Reply:
(and elsewhere)

Reply:
That broke the finding and sorting. Took a while to figure out but it's working now.